conans/client/tools/settings.py

```diff
@@ -0,0 +1,62 @@
+from conans.errors import ConanInvalidConfiguration
+from conans.client.build.cppstd_flags import cppstd_flag, cppstd_from_settings
+from conans.client.tools.oss import OSInfo
+
+
+def check_min_cppstd(conanfile, cppstd, gnu_extensions=False):
+    """ Check if current cppstd fits the minimal version required.
+
+        In case the current cppstd doesn't fit the minimal version required
+        by cppstd, a ConanInvalidConfiguration exception will be raised.
+
+    :param conanfile: ConanFile instance with cppstd to be compared
+    :param cppstd: Minimal cppstd version required
```

Comment:
Can we use for `cppstd` argument a value other than an integer? Does it make sense to require `check_min_cppstd(conanfile, cppstd="gnu17", gnu_extensions=False)`?

Reply:
it's acceptable, but it does not affect the comparison. All values passed passed by argument have the "gnu" prefix remove before.

Reply:
You can take a look on the test `test_check_min_cppstd_from_settings` it validates this case.

Reply:
If people write:

```
check_min_cppstd(conanfile, cppstd="gnu17")  # default 'gnu_extension' is False
```

instead of

```
check_min_cppstd(conanfile, cppstd="17", gnu_extension=True)
```

will the first work for 'compiler.cppstd=17`?

Reply:
if I have:
`check_min_cppstd(conanfile, cppstd="gnu17")`
`conanfile.settings.compiler.cppstd="17"`

It will work, because `gnu17` will become `17`, which is >= 17

But, if I have:

`check_min_cppstd(conanfile, cppstd="17", gnu_extension=True)`
`conanfile.settings.compiler.cppstd="17"`

It will not work. As gnu is mandatory and compiler.cppstd="17" does not support it, it will raise an error? 

Sounds a bit confusing, no?

Reply:
IMO, although  we will document it (only the `integer` part of the standard, no `gnu17`, neither `C++17`,...) I think we should make this easy check and raise if the input is invalid. Do we expect always an integer?

Reply:
> IMO, although we will document it 

totally! When created the first doc draft, it didn't have the gnu_extension parameter. I need to update it.

> Do we expect always an integer?

No, e.g. "gnu17" is valid.

Reply:
@jgsogo please, take a look https://github.com/conan-io/docs/pull/1467

Reply:
> No, e.g. "gnu17" is valid.

`cppstd="gnu17"` doesn't raise, but **it won't work as the user expect**. I would be exactly the same as providing `cppstd="17"`. That's what I say. I'm wondering if we should check that the `cppstd` argument should be just the C++ standard and forbid any extra character.

Reply:
do you mean, only accepting integer values?

Reply:
Yes, we've been talking about it and **we want to check here that the input is just an integer** (add some message to help the user).

Reply:
Done! I've inserted a new assert to check cppstd type